manifests/init.pp

```diff
@@ -48,94 +42,100 @@
-  $package_name          = $openvmtools::params::package_name,
-  $desktop_package_name  = $openvmtools::params::desktop_package_name,
-  $service_ensure        = 'running',
-  $service_name          = $openvmtools::params::service_name,
-  $service_enable        = true,
-  $service_hasstatus     = $openvmtools::params::service_hasstatus,
-  $service_pattern       = 'vmtoolsd',
-) inherits openvmtools::params {
-
-  $supported = $openvmtools::params::supported
+  Enum['absent','present']            $ensure                = 'present',
+  Boolean                             $autoupgrade           = false,
+  String[1]                           $desktop_package_name  = 'open-vm-tools-desktop',
+  Boolean                             $manage_epel           = false,
+  String[1]                           $package_name          = 'open-vm-tools',
+  Stdlib::Ensure::Service             $service_ensure        = 'running',
+  Variant[String[1],Array[String[1]]] $service_name          = 'vmtoolsd',
+  Boolean                             $service_enable        = true,
+  Boolean                             $service_hasstatus     = true,
+  String[1]                           $service_pattern       = 'vmtoolsd',
+  Boolean                             $with_desktop          = false,
+) {
 
-  # Validate our booleans
-  validate_bool($with_desktop)
-  validate_bool($autoupgrade)
-  validate_bool($service_enable)
-  validate_bool($service_hasstatus)
-  validate_bool($supported)
+  $supported = lookup('openvmtools::supported', Boolean, 'first', false)
 
-  case $ensure {
-    /(present)/: {
-      if $autoupgrade == true {
-        $package_ensure = 'latest'
-      } else {
-        $package_ensure = 'present'
-      }
 
-      if $service_ensure in [ running, stopped ] {
-        $service_ensure_real = $service_ensure
-      } else {
-        fail('service_ensure parameter must be running or stopped')
-      }
-    }
-    /(absent)/: {
-      $package_ensure = 'absent'
-      $service_ensure_real = 'stopped'
-    }
-    default: {
-      fail('ensure parameter must be present or absent')
+  if $ensure == 'present' {
+    $package_ensure = $autoupgrade ? {
+      true    => 'latest',
+      default => 'present',
     }
+    $service_ensure_real = $service_ensure
+  }
+  else {  # ensure == 'absent'
+    $package_ensure = 'absent'
+    $service_ensure_real = 'stopped'
   }
 
-  case $::virtual {
-    'vmware': {
-      if $supported {
-        package { $package_name :
-          ensure => $package_ensure,
-        }
-        if $with_desktop {
-          package { $desktop_package_name :
-            ensure => $package_ensure,
-          }
-        }
-
-        service { $service_name :
-          ensure    => $service_ensure_real,
-          enable    => $service_enable,
-          hasstatus => $service_hasstatus,
-          pattern   => $service_pattern,
-          require   => Package[$package_name],
+  if $facts.get('virtual') == 'vmware' {
```

Comment:
`get()` is only available in Puppet 6 yet this module states that it supports Puppet versions below 6. Please change this and the other instances of `.get()` so it supports versions below Puppet 6.

```
# metadata.json
    {
      "name": "puppet",
      "version_requirement": ">=2.7.20 <7.0.0"
    },
```